archive/xar_header.ksy

```diff
@@ -0,0 +1,58 @@
+meta:
+  id: xar_header
+  title: eXtensible ARchiver
+  file-extension:
+    - pkg
+    - xar
+    - xip
+  xref:
+    mime: application/x-xar
+    pronom: fmt/600
+    wikidata: Q1093556
+  license: CC0-1.0
+  ks-version: 0.9
+  encoding: UTF-8
+  endian: be
+doc: |
+  From Wikipedia:
+
+  "XAR (short for eXtensible ARchive format) is an open source file archiver
+  and the archiver’s file format. It was created within the OpenDarwin project
+  and is used in macOS X 10.5 and up for software installation routines, as
+  well as browser extensions in Safari 5.0 and up."
+
+  It should be noted that there is a different version from Apple uses the
+  same version number as the (unmaintained) version described here, but that
+  supports additional checksums, but doesn't have checksum_name.
+versions
+doc-ref: https://github.com/mackyle/xar/wiki/xarformat
+seq:
+  - id: magic
+    contents: 'xar!'
+  - id: header_size
+    type: u2
+  - id: version
+    type: u2
+    valid: 1
+  - id: len_toc_compressed
+    -orig-id: toc_length_compressed
+    type: u8
+  - id: toc_length_uncompressed
+    type: u8
+  - id: checksum_algorithm
+    type: u4
+    enum: checksum_algorithms
+  - id: checksum_name
+    type: strz
+    size: 36
+    if: checksum_algorithm == checksum_algorithms::other
+  - id: toc
```

Comment:
is it a `str`?

Reply:
The XML you mean? It is zlib compressed data. Uncompressed it should be a str or strz

Reply:
Then it is a strz.

Reply:
> Then it is a strz.

Is the field really supposed to be zero-terminated? If not, I think it would be better to declare it as just `str`. That way if the data does contain a zero byte for some reason, you'll get an error when you parse the XML data (zero bytes are disallowed in XML). If you declare it as `strz` even though it isn't, the data will instead get silently truncated if a zero byte appears.

Also, are you sure that the XML data always has a fixed encoding? XML can also be encoded as UTF-16, not just UTF-8. So unless the XAR format requires one specific encoding for the XML data, I would parse it as raw bytes and let the XML parser handle the encoding.

Reply:
I looked at the source code of two versions of the program (the one on GitHub and the one from Apple) and it is explicitly set to UTF-8.